canard.c

```diff
@@ -151,29 +151,20 @@ int16_t canardBroadcast(CanardInstance* ins,
     uint32_t can_id = 0;
     uint16_t crc = 0xFFFFU;
 
-    if (canardGetLocalNodeID(ins) == 0)
+    can_id = ((uint32_t) priority << 26U) | ((uint32_t) data_type_id << 8U) | ((uint32_t) canardGetLocalNodeID(ins) << 1U);
+
+    if (canardGetLocalNodeID(ins) == CANARD_BROADCAST_NODE_ID) // Anonymous message transfer
     {
         if (payload_len > 7)
         {
             return -CANARD_ERROR_NODE_ID_NOT_SET;
         }
 
-        static const uint16_t DTIDMask = (1U << ANON_MSG_DATA_TYPE_ID_BIT_LEN) - 1U;
-
-        if ((data_type_id & DTIDMask) != data_type_id)
-        {
-            return -CANARD_ERROR_INVALID_ARGUMENT;
-        }
-
-        // anonymous transfer, random discriminator
-        const uint16_t discriminator = (uint16_t)((crcAdd(0xFFFFU, payload, payload_len)) & 0x7FFEU);
-        can_id = ((uint32_t) priority << 24U) | ((uint32_t) discriminator << 9U) |
-                 ((uint32_t) (data_type_id & DTIDMask) << 8U) | (uint32_t) canardGetLocalNodeID(ins);
+        uint8_t pseudo_id = crcAdd(0xFFFFU, payload, payload_len) & 0x7FU;
+        can_id |= (1U << 24U) | ((uint32_t) pseudo_id << 1U);
```

Comment:
`pseudo_id` does not affect the final value of `can_id`. Look at the branching logic, when we end up here, the bits 8...1 are already set to 0xFF.